src/data/history.ts

```diff
@@ -185,16 +185,12 @@ export const fetchDate = (
   hass: HomeAssistant,
   startTime: Date,
   endTime: Date,
-  entityId?: string
+  entityIds: string[]
 ): Promise<HassEntity[][]> =>
   hass.callApi(
     "GET",
     `history/period/${startTime.toISOString()}?end_time=${endTime.toISOString()}&minimal_response${
-      entityId ? `&filter_entity_id=${entityId}` : ``
-    }${
-      entityId && !entityIdHistoryNeedsAttributes(hass, entityId)
```

Comment:
Realistically its quite hard to determine if we need to fetch attributes or not when querying for a collection. At most I think I _could_ run through all of the entity ids and if none of them need attributes not retrieve the attributes, but I am actually not too certain, so any guidance is appreciated.

Reply:
You'll need to set the flag to true if any entity needs it in the query since it will tell the backend to not pull the data in the query as its not smart enough to union two queries together for ones that need attributes and one that don't so if any entity needs it you can't pass no_attributes.

Its possible to optimize it but we have so much branching logic on the backend to handle old database formats its not something I want to do right now

Reply:
I'm not fully certain I understand, are you saying that right now it won't work correctly because I haven't set the no_attributes flag, or that it will be an inefficient call when no attributes are required?

Reply:
It will be inefficient if you don't set no_attributes because it has to select a lot of data from the database, but you can't set it if any of the selected entities require attributes